DS13/code/modules/borg/items+structures.dm

```diff
@@ -0,0 +1,557 @@
+	anchored = TRUE
+	can_buckle = TRUE
+	can_be_unanchored = FALSE
+	max_buckled_mobs = TRUE
+	buildstacktype = null
+	item_chair = null // if null it can't be picked up
+	var/restrained = FALSE //can they unbuckle easily?
+
+
+/obj/structure/chair/borg/conversion/user_buckle_mob(mob/living/carbon/human/M, mob/user)
+	if(!isBorgDrone(M))
+		to_chat(M, "Assimilate them first")
+		return
+	M.unequip_everything()
+	icon_state = "borg_off"
+	visible_message("parts of [src] start to shift and move")
+	playsound(loc, 'DS13/sound/effects/borg/machines/convert_table.ogg', 50, 1, -1)
+	var/image/armoverlay = image('DS13/icons/borg/borg.dmi')
+	armoverlay.icon_state = "borg_arms"
+	armoverlay.layer = ABOVE_MOB_LAYER
+	overlays += armoverlay
+	var/image/armoroverlay = image('DS13/icons/borg/borg.dmi')
+	armoroverlay.icon_state = "borgarmour"
+	armoroverlay.layer = ABOVE_MOB_LAYER
+	overlays += armoroverlay
+	sleep(35)
+	playsound(loc, 'DS13/sound/effects/borg/machines/convert_table2.ogg', 50, 1, -1)
+	icon_state = "borg_off"
+	M.equipOutfit(/datum/outfit/borg, visualsOnly = FALSE) //Outfit handles name etc.
+	cut_overlays()
+	overlays -= armoverlay
+	overlays -= armoroverlay
+	qdel(armoroverlay)
+	qdel(armoverlay)
+	return TRUE
+
+
+/obj/structure/chair/borg/charging
+	name = "recharging alcove"
+	desc = "We must recharge to regain"
+	icon_state = "borgcharger"
+	icon = 'DS13/icons/borg/borg.dmi'
+	anchored = TRUE
+	can_buckle = TRUE
+	can_be_unanchored = FALSE
+	max_buckled_mobs = 1
+	resistance_flags = FIRE_PROOF
+	buildstacktype = null
+	item_chair = null // if null it can't be picked up
+	var/cooldown2 = 110 //music loop cooldowns
+	var/saved_time2 = 0
+	var/sound = 'DS13/sound/effects/borg/machines/alcove.ogg'
+	var/mob/living/carbon/human/user
+
+/obj/structure/chair/borg/charging/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/structure/chair/borg/charging/process()
+	if(user)
+		user.adjustBruteLoss(-3)
+		user.adjustFireLoss(-3)
+		user.adjustOxyLoss(-3)
+		if(user.stat == DEAD)
+			user.updatehealth() // Previous "adjust" procs don't update health, so we do it manually.
+			user.set_heartattack(FALSE)
+			user.revive()
+			user.adjustBruteLoss(-20) //give them a real kick so they do actually revive
+			user.adjustFireLoss(-20)
+			user.adjustOxyLoss(-20)
+			user.adjustToxLoss(-20)
+			user.restoreEars()
+			user.adjust_eye_damage(-5)
+			if(user.eye_blind || user.eye_blurry)
+				user.set_blindness(0)
+				user.set_blurriness(0)
+			if(user.blood_volume < BLOOD_VOLUME_NORMAL)
+				user.blood_volume += 5 // regenerate blood rapidly
+
+	if(world.time >= saved_time2 + cooldown2)
+		saved_time2 = world.time
+		playsound(src,sound,10,0)
+	else
+		return
+
+/obj/structure/chair/borg/charging/user_buckle_mob(mob/living/M, mob/User)
+	if(ishuman(M) && M.loc == loc)
+		var/mob/living/carbon/human/H = M
+		if(isBorgDrone(H))
+			user = H
+			to_chat(H, "<span class='warning'>Establishing connection...</span>")
```

Comment:
What if this person doesnt have a client? Will it silently fail or runtime?

Reply:
Good point